dnscrypt-proxy/xtransport.go

```diff
@@ -447,7 +447,7 @@ func (xTransport *XTransport) resolveAndUpdateCache(host string) error {
 			}
 		}
 	}
-	if err != nil && xTransport.ignoreSystemDNS {
+	if err != nil && !xTransport.ignoreSystemDNS {
```

Comment:
If `ignoreSystemDNS` is `false`, the system DNS is already tried just above.

Here, this is only if it is `true`, but for some reason, the configured bootstrap resolvers didn't work. Hence the warning message because this is not supposed to happen, and maybe the configuration needs to be changed.